src/components/MiningPositionList/index.js

```diff
@@ -0,0 +1,263 @@
+import { AutoColumn } from '../Column'
+import { RowFixed } from '../Row'
+import { ButtonLight } from '../ButtonStyled'
+import { TYPE } from '../../Theme'
+import FormattedName from '../FormattedName'
+
+dayjs.extend(utc)
+
+const PageButtons = styled.div`
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  margin-top: 2em;
+  margin-bottom: 0.5em;
+`
+
+const Arrow = styled.div`
+  color: ${({ theme }) => theme.primary1};
+  opacity: ${props => (props.faded ? 0.3 : 1)};
+  padding: 0 20px;
+  user-select: none;
+  :hover {
+    cursor: pointer;
+  }
+`
+
+const List = styled(Box)`
+  -webkit-overflow-scrolling: touch;
+`
+
+const DashGrid = styled.div`
+  display: grid;
+  grid-gap: 1em;
+  grid-template-columns: 5px 0.5fr 1fr;
+  grid-template-areas: 'number name uniswap';
+  align-items: flex-start;
+  padding: 20px 0;
+
+  > * {
+    justify-content: flex-end;
+    width: 100%;
+
+    :first-child {
+      justify-content: flex-start;
+      text-align: left;
+      width: 20px;
+    }
+  }
+
+  @media screen and (min-width: 1200px) {
+    grid-template-columns: 35px 2.5fr 1fr;
+    grid-template-areas: 'number name uniswap';
+  }
+
+  @media screen and (max-width: 740px) {
+    grid-template-columns: 2.5fr 1fr;
+    grid-template-areas: 'name uniswap';
+  }
+
+  @media screen and (max-width: 500px) {
+    grid-template-columns: 2.5fr 1fr;
+    grid-template-areas: 'name uniswap';
+  }
+`
+
+const ListWrapper = styled.div``
+
+const ClickableText = styled(Text)`
+  color: ${({ theme }) => theme.text1};
+  &:hover {
+    cursor: pointer;
+    opacity: 0.6;
+  }
+
+  text-align: end;
+  user-select: none;
+`
+
+const DataText = styled(Flex)`
+  align-items: center;
+  text-align: center;
+  color: ${({ theme }) => theme.text1};
+  & > * {
+    font-size: 1em;
+  }
+
+  @media screen and (max-width: 600px) {
+    font-size: 13px;
+  }
+`
+
+const SORT_FIELD = {
+  VALUE: 'VALUE',
+  UNISWAP_RETURN: 'UNISWAP_RETURN'
+}
+
+function MiningPositionList({ miningPositions }) {
+  // const below500 = useMedia('(max-width: 500px)')
+  const below740 = useMedia('(max-width: 740px)')
+
+  // pagination
+  const [page, setPage] = useState(1)
+  const [maxPage, setMaxPage] = useState(1)
+  const ITEMS_PER_PAGE = 10
+
+  // sorting
+  const [sortDirection, setSortDirection] = useState(true)
+  const [sortedColumn, setSortedColumn] = useState(SORT_FIELD.VALUE)
+
+  useEffect(() => {
+    setMaxPage(1) // edit this to do modular
+    setPage(1)
+  }, [miningPositions])
+
+  useEffect(() => {
+    if (miningPositions) {
+      let extraPages = 1
+      if (miningPositions.length % ITEMS_PER_PAGE === 0) {
+        extraPages = 0
+      }
+      setMaxPage(Math.floor(miningPositions.length / ITEMS_PER_PAGE) + extraPages || 1)
+    }
+  }, [miningPositions])
+
+  const ListItem = ({ miningPosition, index }) => {
+    const pairPercentage = miningPosition.balance / 1e18 / miningPosition.pairData.totalSupply
```

Comment:
Could this be adjusted on the subgraph so clients do have to do the divison?

Reply:
the `miningPosition.balance` and `pairData.totalSupply` are coming from 2 different sub graphs so the division had to be on the client

Reply:
I guess im more specifically wondering about "miningPosition.balance / 1e18" - could the "balance" field on the mining subgraph be adjusted for decimals?